README.md

```diff
@@ -167,7 +170,32 @@ retrieve them through the command `CLUSTER SLOTS` at runtime.
 * `pool_size`: Number of connected clients to each Redis node. Default: `10`
 * `pool_max_overflow`: Max number of extra clients that can be started when the pool is exhausted. Default: `0`
 * `password`: Password to use for a Redis cluster configured with `requirepass`. Default: `""` (i.e. AUTH not sent)
-* `socket_options`: Extra socket [options](http://erlang.org/doc/man/gen_tcp.html#type-option). Enables selecting host interface or perf. tuning. Default: `[]`
+* `socket_options`: Extra socket [options](http://erlang.org/doc/man/gen_tcp.html#type-option). Enables selecting host interface or perf. tuning. Default: `[]`.
+   However, there are some options added by default by the underlaying eredis
+   client. We suggest the following socket options, in order to efficiently
+   detect a failing connection and trigger an update of the cluster topology in
+   case of failover and other Redis Cluster events:
+
+   * `{send_timeout, Timeout}` and `{send_timeout_close, true}`: Makes sure send
+     errors are not ignored. Triggers a reconnect on send errors.
+   * `{nodelay, true}`: Send TCP packets immediately. A Redis command is sent as
+     a whole, not in pieces, so there is no point delaying the send.
+   * `{keepalive, true}`: Enables TCP keepalive. Enabled by default on Linux.
+   * Raw Linux-specific options, to detect a failing socket as soon as possible:
+
+     ```Erlang
+     %% See Linux kernel file: include/uapi/linux/tcp.h
+     %%  - send keep-alive packets after 1s (minimum possible) of no traffic
+     %%  - time out after one failed try if the keep-alive is not acknowledged
+     %% TCP_KEEPIDLE in seconds - cannot be more granular here:
+     {raw, 6, 4, <<1:32/native>>},
+     %% TCP_KEEPCNT - try only once, fail fast, better than hanging:
+     {raw, 6, 6, <<1:32/native>>},
+     %% TCP_KEEPINTVL - probably does not matter when KEEPCNT == 1:
```

Comment:
> probably does not matter when KEEPCNT == 1

I was wrong in this, better remove that part of the comment.

Reply:
What should I write here instead? It's good to have some explanation of each raw options.

Reply:
I used this, more or less exactly. Looks OK now?